cunumeric/module.py

```diff
@@ -1460,17 +1460,30 @@ def check_shape_dtype(
             f"All arguments to {func_name} "
             "must have the same number of dimensions"
         )
-    if ndim > 1 and _builtin_any(
-        shape[1:axis] != inp.shape[1:axis]
-        and shape[axis + 1 :] != inp.shape[axis + 1 :]
-        for inp in inputs
-    ):
-        raise ValueError(
-            f"All arguments to {func_name} "
-            "must have the same "
-            "dimension size in all dimensions "
-            "except the target axis"
-        )
+
+    if func_name == "stack":
```

Comment:
I'm really not sure about coupling these highly function-specific checks in this general function. AFAIK originally `func_name` was only passed in order to afford more detailed error messages. I think very specific checks that only apply to a single function should just go in that function instead of complicating things here.

Reply:
Per your suggestion, I split original check_shape_dtype into 2 functions. It works really better.